src/shaders/globe_raster.fragment.glsl

```diff
@@ -1,9 +1,40 @@
 uniform sampler2D u_image0;
 varying vec2 v_pos0;
 
+#ifndef FOG
+uniform vec3 u_frustum_tl;
+uniform vec3 u_frustum_tr;
+uniform vec3 u_frustum_br;
+uniform vec3 u_frustum_bl;
+uniform highp vec3 u_globe_pos;
+uniform highp float u_globe_radius;
+uniform vec2 u_viewport;
+#endif
+
 void main() {
-    vec4 color = texture2D(u_image0, v_pos0);
+#if defined(CUSTOM_ANTIALIASING) && defined(USE_STANDARD_DERIVATIVES) && !defined(GLOBE_POLES)
+    vec2 uv = gl_FragCoord.xy / u_viewport;
+
+    vec3 ray_dir = mix(
+        mix(u_frustum_tl, u_frustum_tr, uv.x),
+        mix(u_frustum_bl, u_frustum_br, uv.x),
+        1.0 - uv.y);
+        
+    vec3 dir = normalize(ray_dir);
+
+    vec3 closest_point = dot(u_globe_pos, dir) * dir;
+    float norm_dist_from_center = 1.0 - length(closest_point - u_globe_pos) / u_globe_radius;
 
+    const float antialias_pixel = 4.0;
+    float antialias_factor = antialias_pixel * fwidth(norm_dist_from_center);
+    float antialias = smoothstep(0.0, antialias_factor, norm_dist_from_center);
```

Comment:
Since the antialiasing _eats up_ a bit of the visible globe, we may have some artifacts that appear when the color of the atmosphere isn't fully filling the background space. A very simple fix for that is to update in `atmosphere.fragment.glsl` the check that determines whether you're in the globe or not, and make it slightly more inclusive, refer https://github.com/mapbox/mapbox-gl-js/commit/836fd715dd2b61d11f8b74bfc8c2714c866f5c99#diff-71ed14636d967c45101e9578de4b9280802da8e8d129d9d5a72e93946ea1e77aR46

```diff
-    if (norm_dist_from_center < 1.0) {
+    // Compare against 0.99 instead of 1.0 to give enough room for the custom
+    // antialiasing that might be applied from globe_raster.fragment.glsl
+    if (norm_dist_from_center < 0.99) {
        gl_FragColor = u_color;
        return;
    }
```